sktime/classification/early_classification/_teaser.py

```diff
@@ -0,0 +1,560 @@
+        # minimum difference to the predicted probability
+        X_oc = self._generate_one_class_features(X, preds, probas)
+
+        return X_oc, probas, preds
+
+    def _generate_one_class_features(self, X, preds, probas):
+        # create data set for the one class classifier using predicted probas with the
+        # minimum difference to the predicted probability
+        X_oc = np.hstack((probas, np.ones((len(X), 1))))
+        for i in range(len(X)):
+            for n in range(self.n_classes_):
+                if n != preds[i]:
+                    X_oc[i][self.n_classes_] = min(
+                        X_oc[i][self.n_classes_], X_oc[i][preds[i]] - X_oc[i][n]
+                    )
+        return X_oc
+
+    def _predict_oc_classifier_n_timestamps(
+        self,
+        estimator_preds,
+        X_oc,
+        n_consecutive_predictions,
+        last_idx,
+        next_idx,
+        state_info=None,
+    ):
+        # a List containing the state info for case, edited at each time stamp.
+        # contains 1. the index of the time stamp, 2. the number of consecutive
+        # positive decisions made, and 3. the prediction made
+        if state_info is None:
+            state_info = np.zeros((len(estimator_preds[0]), 3), dtype=int)
+
+        # only compute new indices
+        for i in range(last_idx, next_idx):
+            finished, state_info = self._predict_oc_classifier(
+                X_oc[i - last_idx],
+                n_consecutive_predictions,
+                i,
+                estimator_preds[i - last_idx],
+                state_info,
+            )
+
+        return state_info, finished
+
+    def _predict_oc_classifier(
+        self, X_oc, n_consecutive_predictions, idx, estimator_preds, state_info
+    ):
+
+        # stores whether we have made a final decision on a prediction, if true
+        # state info won't be edited in later time stamps
+        finished = state_info[:, 1] >= n_consecutive_predictions
+        n_instances = len(X_oc)
+
+        last_time_stamp = idx == len(self._classification_points) - 1
+        if last_time_stamp:
+            decision_needed = np.ones(n_instances, dtype=bool)
+        elif self._one_class_classifiers[idx] is not None:
+            offsets = np.argwhere(finished == 0).flatten()
+            decision_needed = np.ones(n_instances, dtype=bool)
+            if len(offsets) > 0:
+                decisions_subset = (
+                    self._one_class_classifiers[idx].predict(X_oc[offsets]) == 1
+                )
+                decision_needed[offsets] = decisions_subset
+
+        else:
+            decision_needed = np.zeros(n_instances, dtype=bool)
+
+        # record consecutive class decisions
+        state_info = np.array(
+            [
+                self._update_state_info(
+                    decision_needed, estimator_preds, state_info, i, idx
+                )
+                for i in range(n_instances)
+            ]
+        )
```

Comment:
Do we want to update state_info for cases we have already made a decision on? I feel like we lose useful information doing so.

Reply:
Oh, you are right: The problem is, that if we add 
`if finished[i] else state_info[i]`

we get the error:

> ValueError: All state_info input instances must be from the previous classification point series length.

Reply:
Hmm, I was wrong about that. We do not get the error